io/eolearn/io/geometry_io.py

```diff
@@ -28,52 +31,48 @@
 class _BaseVectorImportTask(EOTask, metaclass=abc.ABCMeta):
     """Base Vector Import Task, implementing common methods"""
 
-    def __init__(self, feature, reproject=True, clip=False, config=None):
+    def __init__(
+        self, feature: FeatureSpec, reproject: bool = True, clip: bool = False, config: Optional[SHConfig] = None
+    ):
         """
         :param feature: A vector feature into which to import data
-        :type feature: (FeatureType, str)
         :param reproject: Should the geometries be transformed to coordinate reference system of the requested bbox?
-        :type reproject: bool, default = True
         :param clip: Should the geometries be clipped to the requested bbox, or should be geometries kept as they are?
-        :type clip: bool, default = False
         :param config: A configuration object with credentials
-        :type config: SHConfig
         """
         self.feature = self.parse_feature(feature, allowed_feature_types=FeatureTypeSet.VECTOR_TYPES)
         self.config = config or SHConfig()
         self.reproject = reproject
         self.clip = clip
 
     @abc.abstractmethod
-    def _load_vector_data(self, bbox):
+    def _load_vector_data(self, bbox: Optional[BBox]):
         """Loads vector data given a bounding box"""
 
-    def _reproject_and_clip(self, vectors, bbox):
+    def _reproject_and_clip(self, vectors: gpd.GeoDataFrame, bbox: Optional[BBox]) -> gpd.GeoDataFrame:
         """Method to reproject and clip vectors to the EOPatch crs and bbox"""
         if not bbox and (self.reproject or self.clip):
             raise ValueError("To clip or reproject vector data, eopatch.bbox has to be defined!")
 
         if self.reproject:
-            vectors = vectors.to_crs(bbox.crs.pyproj_crs())
+            vectors = vectors.to_crs(bbox.crs.pyproj_crs()) if bbox else vectors
 
         if self.clip:
-            bbox_crs = bbox.crs.pyproj_crs()
+            bbox_crs = bbox.crs.pyproj_crs() if bbox else None
             if vectors.crs != bbox_crs:
                 raise ValueError("To clip, vectors should be in same CRS as EOPatch bbox!")
 
-            extent = gpd.GeoSeries([bbox.geometry], crs=bbox_crs)
+            extent = gpd.GeoSeries([bbox.geometry], crs=bbox_crs) if bbox else None
             vectors = gpd.clip(vectors, extent, keep_geom_type=True)
```

Comment:
can you restructure the code-flow to avoid these awkward checks? The new additions make the code even less readable.

Reply:
done